packages/@sanity/schema/src/legacy/Schema.ts

```diff
@@ -42,22 +56,33 @@ function compileRegistry(schemaDef: any) {
  * @beta
  */
 export class Schema {
-  _original: {name: string; types: any[]}
+  _original: {name: string; types: any[]; parent?: Schema}
   _registry: {[typeName: string]: any}
+  _localTypeNames: string[]
 
   static compile(schemaDef: any): Schema {
     return new Schema(schemaDef)
   }
 
   constructor(schemaDef: any) {
     this._original = schemaDef
-    this._registry = compileRegistry(schemaDef)
+
+    const {registry, localTypeNames} = compileRegistry(schemaDef)
+    this._registry = registry
```

Comment:
Am I correct in assuming that `this._registry` contains exactly the same set of types as it would before?

Reply:
Correct! That was intentional so that `getTypeNames()` and `get()` would not break.

Do note that builtin types (e.g. `sanity.fileAsset`) across difference instances of `createSchema` will now return the _same_ instance. I don't think that's breaking (since these are supposed to be immutable), but it could be considered a slight change of behavior.